uk.co.compucorp.civicrm.hrleaveandabsences/CRM/HRLeaveAndAbsences/Service/LeaveRequest.php

```diff
@@ -139,14 +138,28 @@ protected function update($params) {
    */
   private function getErrorMessageForInvalidStatusTransition($params) {
     $leaveStatuses = LeaveRequest::getStatuses();
-    $isOwnRequest = CRM_Core_Session::getLoggedInContactID() === $params['contact_id'];
+    $isOwnRequest = $this->leaveRequestRightsService->currentUserIsLeaveContact($params['contact_id']);
```

Comment:
Please revert to the default implementation `CRM_Core_Session::getLoggedInContactID() === $params['contact_id']` making the `currentUserIsLeaveContact` method public requires adding a test and its not worth it. I didn't realise it was a private method.

Reply:
Np, done.